node/actors/network/src/metrics.rs

```diff
@@ -245,58 +253,58 @@ impl MeteredStreamStats {
             received: 0.into(),
             established: SystemTime::now(),
             peer_addr,
-            sent_buffer: VecDeque::new(),
-            received_buffer: VecDeque::new(),
+            current_minute_sent: 0.into(),
+            previous_minute_sent: 0.into(),
+            current_minute_received: 0.into(),
+            previous_minute_received: 0.into(),
+            current_minute_start: 0.into(),
         }
     }
 
-    fn read(&mut self, amount: u64) {
+    fn read(&self, amount: u64) {
+        self.update_minute();
         self.received.fetch_add(amount, Ordering::Relaxed);
-        self.received_buffer.push_back((Instant::now(), amount));
-        self.cleanup_buffers();
+        self.current_minute_received
+            .fetch_add(amount, Ordering::Relaxed);
     }
 
-    fn wrote(&mut self, amount: u64) {
+    fn wrote(&self, amount: u64) {
+        self.update_minute();
         self.sent.fetch_add(amount, Ordering::Relaxed);
-        self.sent_buffer.push_back((Instant::now(), amount));
-        self.cleanup_buffers();
+        self.current_minute_sent
+            .fetch_add(amount, Ordering::Relaxed);
     }
 
-    fn cleanup_buffers(&mut self) {
-        let one_minute_ago = Instant::now() - Duration::from_secs(60);
+    fn update_minute(&self) {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards")
+            .as_secs();
 
-        while let Some(&(timestamp, _)) = self.sent_buffer.front() {
-            if timestamp < one_minute_ago {
-                self.sent_buffer.pop_front();
-            } else {
-                break;
-            }
-        }
+        let elapsed_secs = now.saturating_sub(self.current_minute_start.load(Ordering::Relaxed));
 
-        while let Some(&(timestamp, _)) = self.received_buffer.front() {
-            if timestamp < one_minute_ago {
-                self.received_buffer.pop_front();
-            } else {
-                break;
-            }
+        if elapsed_secs >= 60 {
```

Comment:
nit: I guess this also would work, but fyi I meant more sth like
```
let now = now().as_secs()/60;
if now>current_minute {
  if current_minute == now-1 {
    prev_minute_sent = current_minute_sent  
  } else {
    prev_minute_sent = 0
  }
  current_minute_sent = 0
  current_minute = now
}
```

Reply:
Seems equivalent, unless we don't receive any packet for more than a minute. But then a 1m window is not appropriate anyway.